cdap-ui/app/hydrator/controllers/create/leftpanel-ctrl.js

```diff
@@ -96,7 +96,15 @@ class HydratorPlusPlusLeftPanelCtrl {
         type: this.LEFTPANELSTORE_ACTIONS.PLUGIN_DEFAULT_VERSION_CHECK_AND_UPDATE
       });
       const defaultVersionMap = this.leftpanelStore.getState().plugins.pluginToVersionMap;
-      this.mySettings.set('plugin-default-version', defaultVersionMap);
+      this.mySettings.get('CURRENT_CDAP_VERSION')
+      .then((defaultCDAPVersion) => {
+        if (this.rVersion.version !== defaultCDAPVersion) {
+          this.mySettings.set('plugin-default-version', {});
+          this.mySettings.set('CURRENT_CDAP_VERSION', this.rVersion.version);
+          return;
+        }
+        this.mySettings.set('plugin-default-version', defaultVersionMap);
```

Comment:
I think all these promises need to be chained since they could overwrite each other?